### PR TITLE
Index GloVe embeddings in Lucene

### PR DESCRIPTION
**Pull request by lintool**
Initial work on playing with word embeddings, using deeplearning4j APIs.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.DS_Store
+.classpath
+.project
+.idea
+target/
+bin/
+eval/trec_eval.9.0.4/
```
Comment (Peilin-Yang) [inline .gitignore]: These ignores are interesting...

**Review by Peilin-Yang — APPROVED** [review]
Overall LGTM for this helloworld App. 
I donot think at this stage we should have too rigorous CR.

**Comment by tteofili** [comment]
Looks good to me, perhaps we can use `WordVectorSerializer#loadStaticModel` (as `loadTxtVectors` is deprecated).
We could also rename the hello world app not to be tight to just GloVe, as far as I can see it should work seamlessly with GloVe and word2vec.

As a follow up we could add an Encoder abstraction to index embeddings as `FloatPoint` and find nearest neighbour words with [Lucene NN search](https://github.com/apache/lucene-solr/blob/master/lucene/sandbox/src/java/org/apache/lucene/document/FloatPointNearestNeighbor.java#L347) (max no. of dimensions is 8, because of BKD-Tree impl).
A basic Encoder impl could use DL4J PCA.

File: .gitignore (hunk 1 above)
Comment (lintool) [reply .gitignore]: :)

**Comment by lintool** [comment]
@tteofili PR welcome! :)

**Pull request merged**